llvm/lib/Target/AMDGPU/SIInstrInfo.h

```diff
@@ -41,6 +41,10 @@ class ScheduleHazardRecognizer;
 static const MachineMemOperand::Flags MONoClobber =
     MachineMemOperand::MOTargetFlag1;
 
+/// Mark the MMO of a load as the last use.
+static const MachineMemOperand::Flags MOLastUse =
```

Comment:
Theoretically we need something to ensure the stores aren't reordered, but it's probably not a practical concern as this runs late enough